xbmc/platform/android/MemUtils.cpp

```diff
@@ -35,13 +35,14 @@ void GetMemoryStatus(MemoryStatus* buffer)
   if (!buffer)
     return;
 
-  long availMem, totalMem;
+  long availMem = 0;
+  long totalMem = 0;
 
   if (CXBMCApp::get()->GetMemoryInfo(availMem, totalMem))
   {
     *buffer = {};
-    buffer->totalPhys = totalMem;
-    buffer->availPhys = availMem;
+    buffer->totalPhys = static_cast<uint64_t>(static_cast<unsigned long>(totalMem));
```

Comment:
Is the double casting actually needed?

Reply:
The key point is convert first from `signed long` to `unsigned long` without changing bit length.

Probably the second cast to`uint64_t` is not necessary but in this case I prefer because it indicates what is intended.

Without any explicit cast compiler is doing implicit `uint64_t` cast and does not work in all cases.

Reply:
Tested with only `static_cast<unsigned long>` and fix is maintained. 

Updated PR and backport.